cpp/src/gandiva/external_c_functions.cc

```diff
@@ -0,0 +1,75 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License
+
+#include <llvm/IR/Type.h>
+
+#include "gandiva/engine.h"
+#include "gandiva/exported_funcs.h"
+
+namespace gandiva {
+// calculate the number of arguments for a function signature
+static size_t GetNumArgs(const FunctionSignature& sig, const NativeFunction& func) {
```

Comment:
Anonymous namespace is better than `static` function.

```cpp
namespace {
size_t GetNumArgs(...) {...}
Result<std::pair<std::vector<llvm::Type*>, llvm::Type*>> MapToLLVMSignature(...) {...}
};
namespace gandiva {
arrow::Status ExternalCFunctions::AddMappings(Engine* engine) const {...}
};

Reply:
Done.